### ISSUE TITLE
Implement item removal and turn skipping for using battle_items in battle

### ISSUE THREAD

**Issue by lczielinski** [uchicago-cs/chiventure#1045]
First, I want to be able to remove items completely from a player's list of items once its quantity reaches 0. Currently, items whose quantity is 0 sit in the player struct without being able to be used. This would involve checking if an item's quantity is 0 after being used, and if it is, removing it from the items linked list in the player struct. Second, I want using a battle item to replace making a move. Currently, a player can use as many items as they want and also make a move, but it should work that a player can either consume an item or make a move in one turn, not both.

**Comment by lczielinski** [comment]
I have created a new branch, battle/item-features, to work on these two features. I created a new function, remove_battle_item(), which takes a combatant and an item and removes the item from the combatant's doubly linked list using linked list logic. It is in the battle_logic file. Within the use_battle_item() function, I check to see if the quantity of an item is 0 after consuming it. If so, I call the remove_battle_item() function to remove the item from the list. I also created 3 new tests for this function, and may add more tests later.

**Comment by lczielinski** [comment]
After looking through the battle_flow file, I found the function battle_flow() which takes in the player's move, uses it on the target, and then calls the give_move() function in order for the enemy to take their turn attacking the players. The issue is that the enemy only attacks the player after the player attacks them first. In order for the player using an item to count as a turn, I will have to refactor this function in some way for the enemy to attack the player when the player uses a move OR an item.

**Comment by lczielinski** [comment]
After communicating with the rest of my team, I will be making this second change in another branch to coordinate better with other members of my team. I will create a pull request for removing an item and begin modifying the battle/cli_integration branch for the second feature.

**Comment by lczielinski** [comment]
After more communication, I decided to combine the two changes in the same pull request. To allow using an item to count as the player's turn, I split up the battle_flow() function (which allows both the player and enemy to make an attack) into battle_flow() and enemy_make_move(), the second of which is called at the end of battle_flow(). Then, I call enemy_make_move() in the battle_command_example.c file after the player uses an item, allowing the enemy to attack the player. I also copied corresponding move print statements after calling enemy_make_move() in the example. This means that a player using an item will use the item first, but will then be attacked by their enemy (which could result in a battle defeat).

**Comment by lczielinski** [comment]
To summarize this issue, I identified two problems with battle_items: the first being that an item is not removed from the player's struct after its quantity has decreased to 0, and the second being that the player using an item does not count as that player's turn. To fix the first issue, I implemented a new function, remove_battle_item(), which removes an item from the player's struct and I call it in use_battle_item() if the item's quantity is 0. To fix the second issue, I split the battle_flow() function in another part, enemy_make_move(), which is called at the end battle_flow() and also in battle_command_example.c after the player uses an item. The deeper issue is that the battle flow is set up so that only making attack moves counts as 'turns'--more structural changes will have to be made to the battle_flow.c file to make it easier to have using items count as turns.
